### PR TITLE
Define default buckets for Jaeger metrics

### LINKED ISSUE #338: Allow for a way to overwrite the default histogram buckets for http metrics
## Is your feature request related to a problem? Please describe

Patron uses the jaeger client for tracing, which also does metrics. The default metrics that it exposes use the following histogram buckets for response times:

```
0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10, +Inf
```

These probably make sense for fast services, but in the case where the requests are slow (for example when you're depending on other service calls that are slow), the metrics are not accurate enough with these values, even misleading:

If a request is 251ms, it goes in the 500ms bucket
If a request is 501ms, it goes in the 1s bucket.

So, it doesn’t give an accurate representation of the data, especially if you then visualise this tools like in Grafana.

For services like this, it would be nice if we can overwrite the default buckets to fit the specific service needs better.

## Describe the solution

Allow a way to set different buckets as defaults, if specified these will be set within Patron, otherwise it will use the defaults.

Either via a builder style, or environment variables.

### PR DESCRIPTION
**Pull request by moraitisk**
Signed-off-by: Konstantinos Moraitis <k.moraitis9@gmail.com>

## Which problem is this PR solving?

Closes https://github.com/beatlabs/patron/issues/338

## Short description of the changes

Introduce a new env var `PATRON_JAEGER_DEFAULT_BUCKETS` for defining a list of values that will be used as the default buckets in Jaeger metrics such as `Timer` and `Histogram` types.

The changes also include an upgrade to the `jaeger-lib` that contains a related bug fix, where the buckets were not registered properly.

### CODE REVIEW

File: trace/trace.go
Comment (mantzas) [inline trace/trace.go:52]: Question: If you pass in an empty list will it work and revert to the default prometheus one?

**Review by mantzas — CHANGES_REQUESTED** [review]

File: trace/trace.go
Comment (moraitisk) [reply trace/trace.go:52]: Hello, that's correct! If you do not set the env var, the default prometheus buckets will be used instead. [reference](https://github.com/prometheus/client_golang/blob/d89cf5af8849fcb6d69f2c16bdf3b910e03d341e/prometheus/histogram.go#L188)

File: go.mod
Comment (peterklijn) [inline go.mod:25]: When do you expect this change to be released officially?

Comment (moraitisk) [reply go.mod:25]: I was waiting for a new release of `jaeger-lib` to open this PR, but the fix has been merged since April. I'll keep an eye on it.

Comment (mantzas) [reply go.mod:25]: @peterklijn this PR? If yes as soon as we can have approvals from the rest of the team. @drakos74 @tpaschalis

**Review by mantzas — APPROVED** [review]

File: go.mod
Comment (tpaschalis) [reply go.mod:25]: I think that's referring to a new jaeger-lib release, since this is pinned on commit `135cf56` on the jaeger-lib repo, right?

I don't think there's any danger of us breaking things here with some `go get` command in the future, so  I believe we can proceed with merging here.

**Review by tpaschalis — APPROVED** [review]
LGTM, nice work, and kudos for also fixing things upstream in jaeger-lib!

File: go.mod
Comment (peterklijn) [reply go.mod:25]: Sorry, I meant when do you expect jaeger-lib 2.4.2 to be released. I noticed your PR has been merged to master, but the version you use in go.mod does not point to an official release.

Comment (mantzas) [reply go.mod:25]: @peterklijn  you have a point. Either we go to an official 2.4.1 is the latest release or leave the existing one as it is. We should not use an unofficial release.

Comment (moraitisk) [reply go.mod:25]: Since we measure request latencies on a `Timer` metric type, as seen [here](https://github.com/jaegertracing/jaeger-client-go/blob/master/rpcmetrics/metrics.go#L38), going with the 2.4.1 version is not enough to close this Patron issue. Should we wait for a new release?

Comment (mantzas) [reply go.mod:25]: Is this unrelated with the existing PR? The current is only about making the buckets configurable. If we need to go to the newest i would wait for the next official release if you don't mind.

Comment (moraitisk) [reply go.mod:25]: This PR aimed to make the buckets configurable for all metric types that use them. i.e. `Histogram`, `Timer`.
Using version 2.4.1, the fix will be applied only to the first type, which does not close the linked Patron issue. 
Let's wait for an official release then - no problem with that.

**Review by mantzas — APPROVED** [review]

File: go.mod
Comment (tpaschalis) [reply go.mod:25]: We've talked this through, and we decided that it's okay to go with the pinned commit for now. 

When a new version of `jaeger-lib` hits the ground, we can just go mod tidy at a later time!

**Pull request merged**
